test/flanders/spec_test.clj

```diff
@@ -112,3 +113,20 @@
   (is (s/valid?
        (fs/->spec (f/set-of (f/set-of f/any-str)) "test-set")
        #{#{"foo"}})))
+
+(deftest sig-spec-test
+  (let [spec-key (fs/->spec (f/sig :parameters [(f/int)]) "foo")]
+    (is (match (s/describe spec-key)
+          (['fspec :args (['cat :a0 _] :seq) :ret _ :fn nil] :seq)
+          true
+
+          _
+          false)))
+
+  (let [spec-key (fs/->spec (f/sig :parameters [(f/int)] :rest-parameter (f/int)) "foo")]
```

Comment:
Could you also test the return spec if specified?

Reply:
I had some trouble figuring out how I could do that which is why I resorted to pattern matching on the description of the spec. I suppose another test I could do is use `s/exercise` to produce some values and check if the types line up.

Reply:
I gave this a try.

```clj
  (let [spec-key (fs/->spec (f/sig :parameters [(f/int)]) "foo")
        fs (map first (s/exercise (s/spec spec-key)))]
     (is (every? (fn [f]
                  (any? (f 1)))
                fs)
        "The generated function executes and returns provided correct input.")

    (is (every?
         (fn [f]
           (try (f "1") false
                (catch AssertionError _ true)))
         fs)
        "Then generated function throws when provided incorrect input."))
```

In the REPL, this works just fine, however, when I run `lein test` it causes this error (giant stack trace omitted).

```
Caused by: java.lang.ClassCastException: class clojure.lang.AFunction$1 cannot be cast to class clojure.lang.MultiFn (clojure.lang.AFunction$1 and clojure.lang.MultiFn are in unnamed module of loader 'app')
```

I spent a bit of time trying to figure out what the problem is but, at this time, I'm convinced spending much more time on would be worthwhile.